### PR TITLE
RoomRecorder

### PR DESCRIPTION
**Pull request by frostbyte73**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -44,3 +44,4 @@ require (
 )
 
 replace github.com/pion/ion-sfu => github.com/livekit/ion-sfu v1.20.5
+replace github.com/livekit/protocol => ../protocol
```
Comment (frostbyte73) [inline go.mod:47]: will remove after protocol PR

File: pkg/service/roomservice.go [hunk 2]
```diff
@@ -219,3 +276,24 @@ func (s *RoomService) writeMessage(ctx context.Context, room, identity string, m
...
+		SubmittedAt: time.Now().UnixNano(),
+		Input:       req.Input,
+		Output:      req.Output,
+	}
+	b, err := proto.Marshal(reservation)
+	if err != nil {
+		return "", err
+	}
+
+	err = s.recorder.ReserveRecorder(ctx, string(b), id)
```
Comment (davidzhao) [inline pkg/service/roomservice.go:293]: seems cleaner to have Recorder to be responsible for serialization. interfaces that use structs seem easier to understand/test.

**Review by davidzhao — APPROVED** [review]
looks good! nice work.

one small nit.

File: pkg/service/roomservice.go (hunk 2 above)
Comment (frostbyte73) [reply pkg/service/roomservice.go:293]: I was getting some error about the proto already being initialized when importing the recorder package from the worker, but it seems ok now. Might've just been out of sync

**Pull request merged**